docs/component/open-attestation.md

```diff
@@ -4,8 +4,311 @@ title: OpenAttestation Wrapper SDK (javascript)
 sidebar_label: Wrapper SDK (javascript)
 ---
 
-Overview
+The OpenAttestation SDK (JavaScript) is a npm module that allows you to manipulate OA documents programmatically. This is useful if you are building your own API or web components. Some common use cases of this module:
+- [Generating OA documents from external data sources programmatically](#wrapping-documents)
+- [Extracting data from OA documents programmatically](#retrieving-document-data)
+- [Programmatic selective disclosure of data fields from issued OA documents](#obfuscating-data)
 
-## Header
+This module does not provide the following functionality:
+- Programmatic verification of OA documents (refer to [Verification SDK (javascript)](/docs/component/oa-verify))
+- Encryption or decryption of OA documents (refer to [Encryption SDK (javascript)](/docs/component/oa-encryption))
+- Programmatic issuance/revocation of document on the Ethereum blockchain
 
-TBD
+## Installation
+
+1. Install [Node.js](https://nodejs.org/en/)
+1. Install the library: `npm install @govtechsg/open-attestation`
+
+## Usage
+
+### Wrapping documents
+
+Before starting to play with the library, create a file name `document.json` with the following content:
+```json
+{
+  "issuers": [
+    {
+      "documentStore": "0xabcdabcdabcdabcdabcdabcdabcdabcdabcdabcd",
+      "name": "University of Blockchain",
+      "identityProof": {
+        "type": "DNS-TXT",
+        "location": "example.com"
+      }
+    }
+  ]
+}
+```
+
+#### Wrapping a single document
+```javascript
+const { wrapDocument } = require("@govtechsg/open-attestation");
+const document = require("./document.json");
+const util = require("util");
+
+const wrappedDocument = wrapDocument(document);
+
+console.log(util.inspect(wrappedDocument, { showHidden: false, depth: null }));
+```
+
+Will display:
+
+```javascript
+{
+  version: 'open-attestation/2.0',
+  data: {
+    issuers: [
+      {
+        documentStore: '5924d910-8916-446a-b1c3-55e2f86dd8f3:string:0xabcdabcdabcdabcdabcdabcdabcdabcdabcdabcd',
+        name: '0f737b20-9e5e-4ec4-bffd-c63002616bfd:string:University of Blockchain',
+        identityProof: {
+          type: '5fc379dd-cd24-4c91-a4c7-cf76f9c96d8d:string:DNS-TXT',
+          location: '62e1c6c3-09a9-4e52-aaf6-9daf24d43657:string:example.com'
+        }
+      }
+    ]
+  },
+  privacy: { obfuscatedData: [] },
+  signature: {
+    type: 'SHA3MerkleProof',
+    targetHash: '956e27c86d4893a971600d328e235ede886f1c00b183257a822667b69886fced',
+    proof: [],
+    merkleRoot: '956e27c86d4893a971600d328e235ede886f1c00b183257a822667b69886fced'
+  }
+}
+```
+Note: 
+- Every time you run `wrapDocument`, it will create different hashes (in front of every field in the data object), `targetHash` and `merkleRoot`.
+- `targetHash` and `merkleRoot` are always the same when wrapping one document.
```

Comment:
Does this mean that only the hashes in front of every field in the data object changes on each run? So `targetHash` and `merkleRoot` never changes when wrapping one doc

Reply:
Nope I wrote the opposite above 
`Every time you run `wrapDocument`, it will create different hashes (in front of every field in the data object), `targetHash` and `merkleRoot`.`

if you wrap one document, targetHash and merkleRoot will always be equivalent, Maybe change same to equivalent so ?

Reply:
Ok was getting confused, perhaps the following will be clearer:
Note:
- Every time you run `wrapDocument`, a salt is generated in front of each field in the data object. `targetHash` and `merkleRoot` also change.
- If you are wrapping one document, `targetHash` and `merkleRoot` are equivalent.

Reply:
updated